src/EnergyPlus/SolarShading.cc

```diff
@@ -359,6 +359,71 @@ void InitSolarCalculations(EnergyPlusData &state)
+    state.dataIPShortCut->rNumericArgs({1, 4}) = 0.0; // so if nothing gotten, defaults will be maintained.
+    state.dataIPShortCut->cAlphaArgs(1) = "";
+    state.dataIPShortCut->cAlphaArgs(2) = "";
+    cCurrentModuleObject = "ShadowCalculation";
+    NumItems = state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, cCurrentModuleObject);
+    NumAlphas = 0;
+    NumNumbers = 0;
+    if (NumItems >= 1) {
+        if (NumItems > 1) {
+            ShowWarningError(state, cCurrentModuleObject + ": More than 1 occurrence of this object found, only first will be used.");
+        }
+        state.dataInputProcessing->inputProcessor->getObjectItem(state,
+                                                                 cCurrentModuleObject,
+                                                                 1,
+                                                                 state.dataIPShortCut->cAlphaArgs,
+                                                                 NumAlphas,
+                                                                 state.dataIPShortCut->rNumericArgs,
+                                                                 NumNumbers,
+                                                                 IOStat,
+                                                                 state.dataIPShortCut->lNumericFieldBlanks,
+                                                                 state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                 state.dataIPShortCut->cAlphaFieldNames,
+                                                                 state.dataIPShortCut->cNumericFieldNames);
+        state.dataSolarShading->ShadowingCalcFrequency = state.dataIPShortCut->rNumericArgs(1);
+    }
+
+    int aNum = 1;
+    unsigned pixelRes = 512u;
+    if (NumAlphas >= aNum) {
+        if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(aNum), "Scheduled")) {
+            state.dataSysVars->shadingMethod = ShadingMethod::Scheduled;
+        } else if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(aNum), "Imported")) {
+            if (state.dataScheduleMgr->ScheduleFileShadingProcessed) {
+                state.dataSysVars->shadingMethod = ShadingMethod::Imported;
+            } else {
+                //
+            }
+        } else if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(aNum), "PolygonClipping")) {
+            state.dataSysVars->shadingMethod = ShadingMethod::PolygonClipping;
+        } else if (UtilityRoutines::SameString(state.dataIPShortCut->cAlphaArgs(aNum), "PixelCounting")) {
+            state.dataSysVars->shadingMethod = ShadingMethod::PixelCounting;
+        } else {
+            //
+        }
+    } else {
+        state.dataSysVars->shadingMethod = ShadingMethod::PolygonClipping;
+    }
+}
+
 void GetShadowingInput(EnergyPlusData &state)
```

Comment:
Finally had to resort to adding a simplified version of input processing just for getting shadow calculation method (which belongs to the solar calculation module) to be used in the surface geometry processing stage, when a full-blown shadow ShadowCalculation input function would be premature and causing some problems (at least in some existing unit tests).

Reply:
Use a new method suggested by the reviewer, now this new function is no longer needed.